Orleans.Sagas/SagaGrain.cs

```diff
@@ -284,11 +288,16 @@ private ActivityContext CreateActivityRuntimeContext(ActivityDefinition definiti
                 grainContext,
                 properties.ToDictionary(x => x.Key, y => y.Value)
             );
-        }
-
-        private void ResumeCompleted()
-        {
-            logger.Info($"Saga {this} has completed with status '{State.Status}'.");
-        }
+        }
+
+        private void ResumeCompleted()
+        {
+            logger.Info($"Saga {this} has completed with status '{State.Status}'.");
+        }
+
+        private void AddActivityError(ActivityDefinition activityDefinition, Exception exception)
+        {
+            State.Errors[activityDefinition.Type] = new SagaError(exception);

```

Comment:
Pay attention that if we have a saga with multiple same activity type, it will problematic to determine which activity is the right not. there is no Id for the activity

Reply:
This is concerning, but is a good compromise for now without more fundamental changes to the model.